### PR TITLE
[receiver/expvarreceiver] Overall structure for new expvarreceiver

### PR DESCRIPTION
**Pull request by jamesmoessis**
**Description:** This PR is the initial Config and Factory implementations for the expvar receiver. There is no receiver implementation yet, it is just a skeleton and basic test for loading config from yaml. 

The `expvarreceiver` intends to collect memstats from Golang applications. The configuration options are collection interval, http client config, and enabling/disabling metrics by name. 

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9592

**Testing:** Basic tests for config validation. More rigorous unit tests will be added when there is more implementation to test. 

**Documentation:**  The README details configuration options and defaults. It doesn't yet go into details on the `metrics` section because the list of metrics names from memstats still needs to be defined by the receiver.

### CODE REVIEW

File: receiver/expvarreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+An Expvar Receiver scrapes metrics from [expvar](https://pkg.go.dev/expvar), 
+which exposes data in JSON format from an HTTP endpoint. The metrics are 
+extracted from the `expvar` variable [memstats](https://pkg.go.dev/runtime#MemStats), 
+which exposes various information about the Go runtime.
+
+> :construction: This receiver is in development and incomplete. It should not be used yet.
+
+## Configuration 
+
+### Default
```
Comment (MovieStoreGuy) [inline receiver/expvarreceiver/README.md:12]: Once we know the list of default metrics we are going to export, we should add them here in that PR with an example of how to disable them if you're not interested in `gc_pause.time` for example.

**Comment by jamesmoessis** [comment]
Hi @dmitryax would you be able to take a look at this PR? 😃

File: receiver/expvarreceiver/Makefile [hunk 2]
```diff
@@ -0,0 +1 @@
+include ../../Makefile.Common
```
Comment (djaglowski) [inline receiver/expvarreceiver/Makefile:1]:
```suggestion
include ../../Makefile.Common

```

File: receiver/expvarreceiver/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+```yaml
+receivers:
+  expvar:
+```
+
+### Customising
+
+The following can be configured:
+- `http_client` - Configure the HTTP client for scraping the expvar variables. The full set of
```
Comment (djaglowski) [inline receiver/expvarreceiver/README.md:25]: This is typically squashed into the config. [Examples](https://github.com/open-telemetry/opentelemetry-collector-contrib/search?q=HTTPClientSettings+%26%26+mapstructure)

File: receiver/expvarreceiver/config.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	if u.Host == "" {
+		return fmt.Errorf("host not found in HTTP endpoint")
+	}
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return fmt.Errorf("scheme must be 'http' or 'https', but was '%s'", u.Scheme)
+	}
+
+	// todo: allow wildcard for enabling/disabling metric names
+	// todo: once all memstat metric names are defined, validate metric names are valid
```
Comment (djaglowski) [inline receiver/expvarreceiver/config.go:55]: I know this component is under development, but todos often stick around unless tracked. Please either remove or create a tracking issue and link from here.

Comment (jamesmoessis) [reply receiver/expvarreceiver/config.go]: Yeah, makes sense. I've removed them.

File: receiver/expvarreceiver/README.md (hunk 3 above)
Comment (jamesmoessis) [reply receiver/expvarreceiver/README.md]: I initially thought that having the HTTP client in it's own config block would be slightly easier to understand, but I also appreciate the need for consistency. I've updated it so it's squashed like in the examples you linked.

**Comment by jamesmoessis** [comment]
RE the changelog checks - I will update it once this PR is approved and ready to be merged. It's the kind of thing that gets a conflict every time I rebase on master because of other things being added to the changelog constantly.

**Review by djaglowski — APPROVED** [review]
LGTM, pending a changelog entry

**Comment by jamesmoessis** [comment]
Thanks for the review @djaglowski! - changelog is updated

**Pull request merged**